src/core/domain.py

```diff
@@ -124,23 +126,29 @@ class DatabaseConnectionInfo:
     dbname: str
 
 
-class ServiceAccountInfo:
-    """Class representing service account and namespace to be used by Kyuubi.
+class SparkServiceAccount(RelationState):
+    """Requirer-side of the Integration Hub relation."""
```

Comment:
This seems correct. What should I update?

Reply:
Maybe it's a nitpick, however on a superficial look, it looked more like a state information class (and hence kept in domain.py) and not exactly an Requirer implementation of the Integration Hub relation. This could potentially be confusing with `IntegrationHubRequirer` which is actually a requirer side implementation of the relation.

Reply:
Well, the integration hub is actually providing service accounts to other charm, so it is not that misleading. Just calling the data object IntegrationHubRequirerData would be more obscure TBH